### PR TITLE
Fixes for wolfSSL conditional porting

### PR DESCRIPTION
**Pull request by dgarske**
# Description

Fix for conditional (signals), which might cause deadlock in high usage situations. 
Added better signal support on MacOS.
Issue created in PR #6437.

# Testing

How did you test?

# Checklist

 - [ ] added tests
 - [ ] updated/added doxygen
 - [ ] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

File: wolfssl/test.h [hunk 1]
```diff
@@ -2237,19 +2223,12 @@ static WC_INLINE void udp_accept(SOCKET_T* sockfd, SOCKET_T* clientfd,
...
-#ifdef COND_NO_REQUIRE_LOCKED_MUTEX
-        THREAD_CHECK_RET(wc_UnLockMutex(&ready->mutex));
-#endif
         THREAD_CHECK_RET(wolfSSL_CondSignal(&ready->cond));
-#ifndef COND_NO_REQUIRE_LOCKED_MUTEX
-        THREAD_CHECK_RET(wc_UnLockMutex(&ready->mutex));
-#endif
-#else
+    #endif
         THREAD_CHECK_RET(wc_UnLockMutex(&ready->mutex));
```
Comment (dgarske) [inline wolfssl/test.h:2230]: Trying to decide if I need to keep the mutex locked or not for the signal. Probably not. If this passes CI I'll try to adjust to unlock the mutex before the signal or wait cond.

File: wolfssl/test.h [hunk 2]
```diff
@@ -695,17 +695,9 @@ static WC_INLINE void srtp_helper_init(srtp_test_helper *srtp)
 static WC_INLINE void srtp_helper_get_ekm(srtp_test_helper *srtp,
                                           uint8_t **ekm, size_t *size)
 {
-#ifndef COND_NO_REQUIRE_LOCKED_MUTEX
-    THREAD_CHECK_RET(wc_LockMutex(&srtp->mutex));
-#endif
-
     if (srtp->server_srtp_ekm == NULL)
         THREAD_CHECK_RET(wolfSSL_CondWait(&srtp->cond, &srtp->mutex));
```
Comment (julek-wolfssl) [inline wolfssl/test.h:699]: Mutex not necessary anymore.

File: wolfssl/test.h (hunk 1 above)
Comment (julek-wolfssl) [reply wolfssl/test.h:2230]: Mutex needs to be locked as per https://linux.die.net/man/3/pthread_cond_signal

>If more than one thread is blocked on a condition variable, the scheduling policy shall determine the order in which threads are unblocked. When each thread unblocked as a result of a pthread_cond_broadcast() or pthread_cond_signal() returns from its call to pthread_cond_wait() or pthread_cond_timedwait(), the thread shall own the mutex with which it called pthread_cond_wait() or pthread_cond_timedwait(). The thread(s) that are unblocked shall contend for the mutex according to the scheduling policy (if applicable), and as if each had called pthread_mutex_lock().
The pthread_cond_broadcast() or pthread_cond_signal() functions may be called by a thread whether or not it currently owns the mutex that threads calling pthread_cond_wait() or pthread_cond_timedwait() have associated with the condition variable during their waits; however, if predictable scheduling behavior is required, then that mutex shall be locked by the thread calling pthread_cond_broadcast() or pthread_cond_signal().

**Review by julek-wolfssl — CHANGES_REQUESTED** [review]
If there is a dedicated mutex just for the condition stuff then I don't see why `lockCount` is needed? Also I would be cautious if the variables being set around the signal calls will register in all threads correctly.

**Comment by dgarske** [comment]
Might fix: https://github.com/curl/curl/issues/11533

**Comment by dgarske** [comment]
Approved. Thanks @julek-wolfssl . Over to @JacobBarthelmeh to finalize.

**Review by JacobBarthelmeh — APPROVED** [review]

**Pull request merged**